Update configure scalar DL document (#174)
scalar-labs/scalar-kubernetes: docs/ConfigureScalarDL.md

## Reference Table

You can refer to the table below to get values to create Kubernetes Secrets and update the configuration files for Scalar DL deployment according to the database you chose.

| Database  | storageType | contactPoints              | username       | password                                 | dynamoBaseResourceUnit | cosmosBaseResourceUnit | 
|:----------|-------------|----------------------------|----------------|------------------------------------------|------------------------|------------------------|
| DynamoDB  | dynamo      | AWS_REGION                 | AWS_ACCESS_KEY | AWS_ACCESS_SECRET_KEY                    | 10                     | N/A                    |
| Cosmos DB | cosmos      | Cosmos DB account endpoint | N/A            | Cosmos DB account primary/secondary key  | N/A                    | 400                    |
| JDBC      | jdbc        | JDBC_CONNECTION_URL        | USERNAME       | PASSWORD                                 | N/A                    | N/A                    |

Note:

* JDBC denotes all relational databases supported by Scalar DL such as 

>   * MySQL, PostgreSQL, Oracle DB and SQL Server

Author: Understood.
Thank you, I will update it.

>   * MySQL, PostgreSQL, Oracle DB and SQL Server

Author: I have updated the sentence and merged it as a single sentence.

## Reference Table

You can refer to the table below to get values to create Kubernetes Secrets and update the configuration files for Scalar DL deployment according to the database you chose.

| Database  | storageType | contactPoints              | username       | password                                 | dynamoBaseResourceUnit | cosmosBaseResourceUnit | 
|:----------|-------------|----------------------------|----------------|------------------------------------------|------------------------|------------------------|
| DynamoDB  | dynamo      | AWS_REGION                 | AWS_ACCESS_KEY | AWS_ACCESS_SECRET_KEY                    | 10                     | N/A                    |
| Cosmos DB | cosmos      | Cosmos DB account endpoint | N/A            | Cosmos DB account primary/secondary key  | N/A                    | 400                    |
| JDBC      | jdbc        | JDBC_CONNECTION_URL        | USERNAME       | PASSWORD                                 | N/A                    | N/A                    |

Note:- 

* JDBC denotes all relational databases supported by Scalar DL such as 
  * MySQL, PostgreSQL, Oracle DB and SQL Server 

>   * AWS RDS for MySQL/PostgreSQL/Oracle/SQL Server

Reviewer: Do we need to specify cloud-based databases separately?

Author: I just mentioned all the supported databases to avoid any confusion.
I will remove the cloud databases.

  * Amazon Aurora MySQL/PostgreSQL
  * Azure Database for MySQL/PostgreSQL

## Create Kubernetes Secrets

Kubernetes Secret is an object that contains a small amount of sensitive data such as a password, a token, or a key.
This optional method is recommended highly for handling credentials in the production environment.
You can refer to the [reference table](#reference-table) section to get the values for `username` and `password` for the database you chose.
You need to create a secret object with `db-username` and `db-password` to store the username and password of the underlying database.

```
kubectl create secret generic scalardl --from-literal db-username=<username> --from-literal db-password=<password>
```

> Note-:

Reviewer: ```suggestion
Note:
```

For Cosmos DB, use the Cosmos DB account name as `username`.

## Update Configuration Files

### Configure schema-loading-custom-values

To create a Scalar DL schema in the database, you need to update the [schema-loading-custom-values.yaml](../conf/schema-loading-custom-values.yaml) file.

> You can refer to the [section](#reference-table) to check which values are needed based on the database used.

Author: I will update here also.

This guide explains how to configure Scalar DL custom values in helm charts for Scalar DL deployment.

> ## Reference Table

Author: I am not sure which one is more suitable here.
I will check and update this.

> ## Reference Table

Author: @scalar-boney 
I have not changed the value for now as I could not confirm which would be more suitable.
We may have to name it `Values` like the document you shared.
What do you think?

> ## Reference Table

Reviewer: I think the heading `Values` is  not enough

> You can refer to the table below to get values to create Kubernetes Secrets and update the [schema-loading-custom-values.yaml](../conf/schema-loading-custom-values.yaml) file according to the database you chose

Author: No, it can be used for all configuration files used.
Thank you for the suggestion. I will update it.

This guide explains how to configure Scalar DL custom values in helm charts for Scalar DL deployment.

> ## Update Configuration Files

Reviewer: Is it correct?
I think the structure is not correct.

Reviewer: Suggestion (Don't follow blindly)

# Configure Scalar DL
## Reference values
## Create kubernetes secrets
## Update configuration files
### Configure schema loading custom value file
### Configure Scalar DL Ledger custom value file
### Configure Scalar DL Auditor custom value file
### Enable Auditor

Author: Thank you for the suggestion.
I will revise it to be more clear.

### Kubernetes Secrets and Schema Loading reference

You can refer to the table below to get values to create Kubernetes Secrets and update the [schema-loading-custom-values.yaml](../conf/schema-loading-custom-values.yaml) file according to the database you chose

> 

Author: Added a table with configuration properties required for each database to create Kubernetes secret and deploy schema loading helm chart.

> | Database  | database | contactPoints              | username          | password                                      | dynamoBaseResourceUnit | cosmosBaseResourceUnit | 

Author: Added a table with configuration properties required for each database to create Kubernetes secret and deploy schema loading helm chart.

> | Database  | database | contactPoints              | username          | password                                      | dynamoBaseResourceUnit | cosmosBaseResourceUnit | 

Author: @scalar-boney 

What about changing `database` to `storageType`?

To create a Scalar DL schema in the database, you need to update the [schema-loading-custom-values.yaml](../conf/schema-loading-custom-values.yaml) file.
You can refer to the [section](#kubernetes-secret-and-schema-loading-reference) to check which values are needed based on the database used.
If any configuration property value is defined as `N/A` in the reference table for your database, you need to remove that property from the configuration file.

**With Kubernetes secrets**

If you have created the `scalardl` Kubernetes secret, you can use the following configuration.

```yaml
schemaLoading:
  existingSecret: scalardl
  
  database: <database>
  contactPoints: <dbContactPoints>
  cosmosBaseResourceUnit: <cosmosBaseResourceUnit>
  dynamoBaseResourceUnit: <dynamoBaseResourceUnit>
  
```

**Without Kubernetes secrets**

If you have not created the `scalardl` Kubernetes secret, you can use the following configuration.

```yaml
schemaLoading:
  database: <database>
  contactPoints: <dbContactPoints>
  username: <username>
  password: <password>
  cosmosBaseResourceUnit: <cosmosBaseResourceUnit>
  dynamoBaseResourceUnit: <dynamoBaseResourceUnit>
```

> ### Ledger and Auditor reference

Author: Added a table with configuration properties required for each database to deploy schema DL and Auditor helm charts.

You can refer the table below to update the [scalardl-custom-values.yaml](../conf/scalardl-custom-values.yaml) and [scalardl-audit-custom-values.yaml](../conf/scalardl-audit-custom-values.yaml) file according to the database of you chose for deployment.

> | Database  | dbStorage | dbContactPoints            | dbUsername        | dbPassword                           |

Reviewer: Ditto

Reviewer: I think we can merge these two lists.
What do you think?

Author: @scalar-boney 

We could merge the two tables.  I will merge and create a new table.
Thank you for the suggestion.

This guide explains how to configure Scalar DL custom values in helm charts for Scalar DL deployment.

## Update Configuration Files

### Kubernetes Secrets and Schema Loading reference

> You can refer to the table below to get values to set create Kubernetes Secrets and update the [schema-loading-custom-values.yaml](../conf/schema-loading-custom-values.yaml) file properties file according to the database you chose for deployment.

Reviewer: This sentence is not much clear `...update the schema-loading-custom-values.yaml file properties file according to the database you chose for deployment.`, please update it.

| Database  | database | contactPoints              | username          | password                             | dynamoBaseResourceUnit | cosmosBaseResourceUnit | 
|:----------|----------|----------------------------|-------------------|--------------------------------------|------------------------|------------------------|
| DynamoDB  | dynamo   | REGION                     | AWS_ACCESS_KEY_ID | AWS_ACCESS_SECRET_KEY                | 10                     | N/A                    |

> | Cosmos DB | cosmos   | Cosmos DB account endpoint | N/A               | Cosmos DB account primary master key | N/A                    | 400                    |

Reviewer: I don't think it supports only the primary master key.

Author: I will add  seconday key along with it.

| JDBC      | jdbc     | JDBC_CONNECTION_URL        | USERNAME          | PASSWORD                             | N/A                    | N/A                    |

### Create Kubernetes Secrets

Kubernetes Secret is an object that contains a small amount of sensitive data such as a password, a token, or a key.
This optional method is highly recommended for handling credentials in the production environment.
You can refer to the [section](#kubernetes-secret-and-schema-loading-reference) to get the values for `username` and `password` for the database you chose.
You need to create a secret object with `db-username` and `db-password` to store the username and password of the underlying database.

```
kubectl create secret generic scalardl --from-literal db-username=<username> --from-literal db-password=<password>
```
Note-:

For Cosmos DB, use the account name as `username` since we can't leave the `db-uername` property empty.

### Configure schema-loading-custom-values

> To create a Scalar DL schema in Cosmos DB, you need to update the [schema-loading-custom-values.yaml](../conf/schema-loading-custom-values.yaml) file.

Reviewer: Is it Cosmos DB specific?

Author: No. I will correct the statement.

You can refer to the [section](#kubernetes-secret-and-schema-loading-reference) to check which values are needed based on the database used.

> If any configuration property value is defined as `N/A` in the reference table for your database, you can remove that property from the configuration file.

Reviewer: ```suggestion
If any configuration property value is defined as `N/A` in the reference table for your database, you need to remove that property from the configuration file.
```

Author: I will update it as per suggestion. Thank you

```yaml
schemaLoading:
  database: <database>
  contactPoints: <dbContactPoints>
  username: <username>
  password: <password>
  cosmosBaseResourceUnit: <cosmosBaseResourceUnit>
  dynamoBaseResourceUnit: <dynamoBaseResourceUnit>
```

### Ledger and Auditor reference

You can refer the table below to update the [scalardl-custom-values.yaml](../conf/scalardl-custom-values.yaml) and [scalardl-audit-custom-values.yaml](../conf/scalardl-audit-custom-values.yaml) file according to the database of you chose for deployment.

| Database  | dbStorage | dbContactPoints            | dbUsername        | dbPassword                           |
|:----------|-----------|----------------------------|-------------------|--------------------------------------|
| DynamoDB  | dynamo    | REGION                     | AWS_ACCESS_KEY_ID | AWS_ACCESS_SECRET_KEY                |

> | Cosmos DB | cosmos    | Cosmos DB account endpoint | N/A               | Cosmos DB account primary master key |

Reviewer: Ditto

| JDBC      | jdbc      | JDBC_CONNECTION_URL        | USERNAME          | PASSWORD                             |

### Configure scalardl-custom-values

> To deploy Scalar DL Ledger on Cosmos DB, you need to update the [scalardl-custom-values.yaml](../conf/scalardl-custom-values.yaml) file.

Reviewer: Ditto

Author: I will correct here also.

You can refer to the [section](#ledger-and-auditor-reference) to check which values are needed based on the database used.

> If any configuration property value is defined as `N/A` in the reference table for your database, you can remove that property from the configuration file.

Reviewer: Is it needed here? I think better to add a note for Cosmos DB.

```yaml
schemaLoading:
  database: <storageType>
  contactPoints: <contactPoints>
  username: <username>
  password: <password>
  cosmosBaseResourceUnit: <cosmosBaseResourceUnit>
  dynamoBaseResourceUnit: <dynamoBaseResourceUnit>
```

### Configure scalardl-custom-values

To deploy Scalar DL Ledger, you need to update the [scalardl-custom-values.yaml](../conf/scalardl-custom-values.yaml) file.

> You can refer to the [section](#reference-table) to check which values are needed based on the database used.

Author: I will update here also.

ledger:
  ....
  scalarLedgerConfiguration:
    dbContactPoints: <contactPoints>
    dbUsername: <username>
    dbPassword: <password>
    dbStorage: <storageType>
```

> Note:-

Reviewer: ```suggestion
Note:
```

For Cosmos DB, you need to remove the `dbUsername` property as it is not required.

### Configure scalardl-audit-custom-values

To deploy the Scalar DL Auditor, you need to update the [scalardl-audit-custom-values.yaml](../conf/scalardl-audit-custom-values.yaml) file.

> You can refer to the [section](#reference-table) to check which values are needed based on the database used.

Reviewer: ditto

Author: I will update here also.

ledger:
  ....
  scalarLedgerConfiguration:
    dbContactPoints: <dbContactPoints>
    dbUsername: <dbUsername>
    dbPassword: <dbPassword>
    dbStorage: <dbStorage>
```

### Configure scalardl-audit-custom-values

> To deploy the Scalar DL Auditor on Cosmos DB, you need to update the [scalardl-audit-custom-values.yaml](../conf/scalardl-audit-custom-values.yaml) file.

Reviewer: Ditto

Author: I will correct it here also.

You can refer to the [section](#ledger-and-auditor-reference) to check which values are needed based on the database used.

> If any configuration property value is defined as `N/A` in the reference table for your database, you can remove that property from the configuration file.

Reviewer: Ditto

Author: I will add a note

auditor:
  ....
  scalarAuditorConfiguration:
    dbContactPoints: <contactPoints>
    dbUsername: <username>
    dbPassword: <password>
    dbStorage: <storageType>
```

> Note:-

Reviewer: ```suggestion
Note:
```

Author: I have corrected the note title to  `Note:`  in all places in which it is used.
Thank you

auditor:
  ....
  scalarAuditorConfiguration:
    dbContactPoints: <dbContactPoints>
    dbUsername: <dbUsername>
    dbPassword: <dbPassword>
    dbStorage: <dbStorage>
```

## Enable Auditor

### Configure scalardl-custom-values

To enable the Auditor service, update the following configurations in the [scalardl-custom-values.yaml](../conf/scalardl-custom-values.yaml) file.

```yaml
scalarLedgerConfiguration:

>   # To use Auditor

Reviewer: Is this format change needed?
